### PR TITLE
OP-1363 | Fix NullPointerException on PatientBillEdit

### PR DESCRIPTION
**Pull request by ArnaudFofou**
See OP-1363.

To fix that issue: 
- I disable button when patient is not selected 
- I enable button when patient is selected

### CODE REVIEW

File: bundle/language_en.properties [hunk 1]
```diff
@@ -1197,6 +1197,7 @@ angal.newbill.payment.btn.key
 angal.newbill.paymentmadebeforebilldate.msg                                                            = Payment made before bill date.
 angal.newbill.paymentreceipt.btn                                                                       = Payment Receipt
 angal.newbill.paymentreceipt.btn.key                                                                   = N
+angal.newbill.pleasefindpatientbeforeremotehim.msg                                                     = Please find patient first before remove him.
```
Comment (dbmalkovsky) [inline bundle/language_en.properties]:
```suggestion
angal.newbill.pleasefindpatientbeforeremotehim.msg                                                     = Please find the patient first before trying to remove him.
```

File: src/main/java/org/isf/accounting/gui/PatientBillEdit.java [hunk 2]
```diff
@@ -846,18 +847,20 @@ private JButton getJButtonTrashPatient() {
...
+					thisBill.setIsPatient(false);
+					thisBill.setPatName(""); //$NON-NLS-1$
+					thisBill.setAdmission(null);
+					// INTERFACE
+					jTextFieldPatient.setText("");
+					jTextFieldPatient.setEditable(false);
+					jButtonPickPatient.setText(MessageBundle.getMessage("angal.newbill.findpatient.btn"));
+					jButtonPickPatient.setToolTipText(MessageBundle.getMessage("angal.newbill.associateapatientwiththisbill.tooltip"));
+				} else {
+					MessageDialog.info(null, MessageBundle.getMessage("angal.newbill.pleasefindpatientbeforeremotehim.msg"));
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/accounting/gui/PatientBillEdit.java]:
```suggestion
					MessageDialog.info(null, "angal.newbill.pleasefindpatientbeforeremotehim.msg");
```
The MessageDIalog methods take the message key, not the resulting text that results from the bundle lookup; that is done by the MessageDialog class.

File: src/main/java/org/isf/accounting/gui/PatientBillEdit.java [hunk 3]
```diff
@@ -846,18 +847,20 @@ private JButton getJButtonTrashPatient() {
...
-				jTextFieldPatient.setText("");
-				jTextFieldPatient.setEditable(false);
-				jButtonPickPatient.setText(MessageBundle.getMessage("angal.newbill.findpatient.btn"));
-				jButtonPickPatient.setToolTipText(MessageBundle.getMessage("angal.newbill.associateapatientwiththisbill.tooltip"));
-				jButtonTrashPatient.setEnabled(false);
+				if (thisBill.getBillPatient() != null) {
+					// BILL
+					thisBill.setBillPatient(null);
+					thisBill.setIsPatient(false);
+					thisBill.setPatName(""); //$NON-NLS-1$
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/accounting/gui/PatientBillEdit.java]:
```suggestion
					thisBill.setPatName("");
```

File: bundle/language_en.properties [hunk 4]
```diff
@@ -1197,6 +1197,7 @@ angal.newbill.payment.btn.key
 angal.newbill.paymentmadebeforebilldate.msg                                                            = Payment made before bill date.
 angal.newbill.paymentreceipt.btn                                                                       = Payment Receipt
 angal.newbill.paymentreceipt.btn.key                                                                   = N
+angal.newbill.pleasefindpatientbeforeremotehim.msg                                                     = Please find the patient first before trying to remove him.
```
Comment (mwithi) [inline bundle/language_en.properties]: I think "him"/"her" is not very professional here, also I don't think the suggestion make sense (no one wants to find a record in order to suddenly remove it).

Why we don't make just the button doing nothing if patient is null?
Also, we can disable the button when no patient is selected

Comment (ArnaudFofou) [reply bundle/language_en.properties]: I remove that message bundle

File: src/main/java/org/isf/accounting/gui/PatientBillEdit.java [hunk 5]
```diff
@@ -845,11 +845,13 @@ private JButton getJButtonTrashPatient() {
...
 			jButtonTrashPatient.setIcon(new ImageIcon("rsc/icons/remove_patient_button.png"));
 			jButtonTrashPatient.setToolTipText(MessageBundle.getMessage("angal.newbill.removethepatientassociatedwiththisbill.tooltip"));
+			if (thisBill.getBillPatient() == null) {
+				jButtonTrashPatient.setEnabled(false);
+			}
 			jButtonTrashPatient.addActionListener(actionEvent -> {
 				// BILL
 				thisBill.setBillPatient(null);
 				thisBill.setIsPatient(false);
-				thisBill.getBillPatient().setCode(0);
```
Comment (mwithi) [inline src/main/java/org/isf/accounting/gui/PatientBillEdit.java:852]: I think the NPE was due only to this line, correct?

**Comment by mwithi** [comment]
Editing a new bill still throws a NPE:

image

```
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException: Cannot invoke "javax.swing.JButton.setEnabled(boolean)" because "this.jButtonTrashPatient" is null
	at org.isf.accounting.gui.PatientBillEdit.setJButtonPickPatient(PatientBillEdit.java:900)
	at org.isf.accounting.gui.PatientBillEdit.getJButtonPickPatient(PatientBillEdit.java:891)
	at org.isf.accounting.gui.PatientBillEdit.getJPanelDate(PatientBillEdit.java:822)
	at org.isf.accounting.gui.PatientBillEdit.getJPanelTop(PatientBillEdit.java:914)
	at org.isf.accounting.gui.PatientBillEdit.initComponents(PatientBillEdit.java:494)
	at org.isf.accounting.gui.PatientBillEdit.<init>(PatientBillEdit.java:383)
	at org.isf.accounting.gui.BillBrowser.lambda$5(BillBrowser.java:529)
```

File: src/main/java/org/isf/accounting/gui/PatientBillEdit.java (hunk 5 above)
Comment (ArnaudFofou) [reply src/main/java/org/isf/accounting/gui/PatientBillEdit.java:852]: yes

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**